Competition2017/src/competition/subsystems/drive/commands/DriveToPositionCommand.java

```diff
@@ -0,0 +1,48 @@
+package competition.subsystems.drive.commands;
+
+import com.google.inject.Inject;
+
+import xbot.common.command.BaseCommand;
+import xbot.common.math.PIDManager;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+import competition.subsystems.drive.DriveSubsystem;
+
+public class DriveToPositionCommand extends BaseCommand{
+
+    private final DriveSubsystem driveSubsystem;
+    private final PIDManager travelManager;
+    
+    private final DoubleProperty distanceTolerance;
+    private double targetPosition;
+    
+    @Inject
+    public DriveToPositionCommand(DriveSubsystem driveSubsystem, XPropertyManager propManager) {
+        this.driveSubsystem = driveSubsystem;
+        this.requires(driveSubsystem);
+        this.travelManager = new PIDManager("DriveToPosition", propManager, 0.1, 0, 0);
+        
+        distanceTolerance = propManager.createPersistentProperty("Distance Tolerance", 0.1);
+    }
+    
+    public void setTargetPosition(double targetPosition) {
+        this.targetPosition = driveSubsystem.convertToInches(targetPosition);
+    }
+    
+    @Override
+    public void initialize() {
+
+    }
+
+    @Override
+    public void execute() {
+        double power = travelManager.calculate(targetPosition, driveSubsystem.getDistance());
+        
+        driveSubsystem.tankDrivePowerMode(power, power);
+    }
+    
+    @Override
+    public boolean isFinished() {
+        return travelManager.isOnTarget(distanceTolerance.get());
```

Comment:
Given that we need to be deliberate about this when reading through it, we should probably append "Ticks" or "Inches" to any properties or variables that contain distance information.

so, "distanceToleranceInches"